UNRELEASED.md

```diff
@@ -10,6 +10,11 @@
          * Some bug fix, see #124
 
      DO NOT LEAVE A BLANK LINE BELOW THIS PREAMBLE -->
+
+### Features
+
+* Type checker: removed `Typing.tla`, `AssumeType`, and `##`, see #518
```

Comment:
Maybe this should go in a `Removed` section?

Just thinking out loud, but we might start using the "keep a changelog" suggested sections: https://keepachangelog.com/en/1.0.0/#how

Reply:
Yep, "Removed" sounds good

Reply:
> Just thinking out loud, but we might start using the "keep a changelog" suggested sections:

Reply:
That is a good idea